cluster/gce/configure-vm.sh

```diff
@@ -242,13 +246,20 @@ EOF
 
 # This should only happen on cluster initialization
 function create-salt-auth() {
-  mkdir -p /srv/salt-overlay/salt/nginx
-  echo "${MASTER_HTPASSWD}" > /srv/salt-overlay/salt/nginx/htpasswd
+  local -r htpasswd_file="/srv/salt-overlay/salt/nginx/htpasswd"
+
+  if [ ! -e ${htpasswd_file} ]; then
+    mkdir -p /srv/salt-overlay/salt/nginx
+    echo "${MASTER_HTPASSWD}" > "${htpasswd_file}"
+  fi
+
+  if [ -e "${KNOWN_TOKENS_FILE}" ]; then
```

Comment:
I like the early-exist approach, but I think it'd actually be better in this case to have all blocks in this function follow the same pattern of `if [ ! -e $file ]; then ...`. That way we have a nice grouping of what's going on in each section, and it's very explicit for someone adding things to the bottom of this function what will be executed (and they won't have to change the logic here). This is a nit, so willing to concede if you feel strongly the other way.